### PR TITLE
Add restangular for app usage

### PR DESCRIPTION
**Pull request by MatiasComercio**
## Summary

- Add grunt-connect-proxy npm dependency
  - Update Gruntfile to use this dependency.
  - API must be served at localhost:7070 during development so as to be
    able to use livereload of angular changes and use a static version
    (war) of the API.
  - Solve Cross Origin request problem with browsers
- Add Students service extending Restangular
- Add Courses service extending Restangular
- Add example of Students and Courses services usage on HomeCtrl
- Add spec utils service
- Add api responses service for testing purposes only (working as
  cassetes for api calls)
- Add all test cases

## Trello Card
https://trello.com/c/zaZ5yBjv/8-configurar-restangular

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -19,14 +21,29 @@ module.exports = function (grunt) { // eslint-disable-line strict
...
+      server: {
+        proxies: [
+          {
+            context: '/api/v1',
+            host: 'localhost',
+            port: 7070,
+            https: false,
+            changeOrigin: true,
+            rewrite: {
+              '^/api/v1': '/grupo1'
```
Comment (MatiasComercio) [inline Gruntfile.js]: This should be updated for current API build. There was a PR that updated this.

File: app/specs/apiResponses.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+// Some of the following code was taken or adapted from:
+// - https://github.com/mgonto/restangular/issues/98 (sanitizeRestangular)
+// - http://www.bradoncode.com/blog/2015/07/25/tip-unit-test-promise-angular/ (promises)
+
```
Comment (MatiasComercio) [inline app/specs/apiResponses.js]: Please remove the above comments: they are not from this code

**Comment by MatiasComercio** [comment]
You also may be handling authentication as it was added by gibarsin in a PR before.
Please update this and then it'll be ready to merge.

File: webapp/src/main/java/ar/edu/itba/paw/webapp/config/WebAuthConfig.java [hunk 3]
```diff
@@ -37,10 +37,10 @@ protected void configure(final HttpSecurity http) throws Exception {
 		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
 				.and()
 					.authorizeRequests()
-					.antMatchers(HttpMethod.POST,"/login").permitAll()
+					.antMatchers(HttpMethod.POST,"/api/v1/login").permitAll()
 					.antMatchers("/**").authenticated()
 				.and()
-					.addFilterBefore(new StatelessLoginFilter("/login", tokenAuthenticationService, userDetailsService, authenticationManager()), UsernamePasswordAuthenticationFilter.class)
+					.addFilterBefore(new StatelessLoginFilter("/api/v1/login", tokenAuthenticationService, userDetailsService, authenticationManager()), UsernamePasswordAuthenticationFilter.class)
```
Comment (MatiasComercio) [inline webapp/src/main/java/ar/edu/itba/paw/webapp/config/WebAuthConfig.java]: @gibarsin please check these changes

Comment (gibarsin) [inline webapp/src/main/java/ar/edu/itba/paw/webapp/config/WebAuthConfig.java]: WebAuthConfig path changes are ok!

File: app/scripts/services/Courses.js [hunk 4]
```diff
@@ -11,6 +11,20 @@ define(['paw', 'restangular'], function(paw) {
           return operation === 'getList' ? data.courses : data;
         }
       );
+      // cannot use RestangularConfigurer.setDefaultHeaders() as it is called
+      // only once and before the token is set
+      // This requests the token each time a request is made
+      // and in this instance it should already be set
+      RestangularConfigurer.addFullRequestInterceptor(
```
Comment (MatiasComercio) [inline app/scripts/services/Courses.js]: Try to delete this duplication

File: app/scripts/services/Students.js [hunk 5]
```diff
@@ -11,6 +11,20 @@ define(['paw', 'restangular'], function(paw) {
           return operation === 'getList' ? data.students : data;
         }
       );
+      // cannot use RestangularConfigurer.setDefaultHeaders() as it is called
+      // only once and before the token is set
+      // This requests the token each time a request is made
+      // and in this instance it should already be set
+      RestangularConfigurer.addFullRequestInterceptor(
```
Comment (MatiasComercio) [inline app/scripts/services/Students.js]: Same

File: app/specs/services/Authentication.spec.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    var AuthenticationService;
+    var expectedToken;
+    var expectedTokenKey = 'ur';
+
+    beforeEach(inject(function(_$rootScope_, _$httpBackend_, _$cookies_,
+      specUtils, apiResponses, Authentication) {
+      $rootScope = _$rootScope_;
+      $httpBackend = _$httpBackend_;
+      $cookies = _$cookies_;
+      specUtilsService = specUtils;
```
Comment (MatiasComercio) [inline app/specs/services/Authentication.spec.js]: Remove this. It's not being used.

**Pull request merged**